data/secscan_model/secscan_v4_model.py

```diff
@@ -0,0 +1,136 @@
+    """
+
+
+class V4SecurityScanner(SecurityScannerInterface):
+    """
+    Implementation of the security scanner interface for Clair V4 API-compatible implementations.
+    """
+
+    def __init__(self, app, instance_keys, storage):
+        self.app = app
+
+        validator = SecurityConfigValidator(
+            app.config.get("FEATURE_SECURITY_SCANNER", False),
+            app.config.get("SECURITY_SCANNER_ENDPOINT"),
+        )
+        if not validator.valid():
+            logger.warning("Failed to validate security scanner V2 configuration")
+            return
+        
+        url_scheme_and_hostname = URLSchemeAndHostname(
+            app.config["PREFERRED_URL_SCHEME"], app.config["SERVER_HOSTNAME"]
+        )
+      
+
+    def load_security_information(self, manifest_or_legacy_image, include_vulnerabilities=False):
+        if not isinstance(manifest_or_legacy_image, Manifest):
+            return None
+
+        status = None
+        try:
+            status = database.ManifestSecurityStatus.get(manifest=manifest_or_legacy_image._db_id)
+        except database.ManifestSecurityStatus.DoesNotExist:
+            return SecurityInformationLookupResult.with_status(
+                ScanLookupStatus.NOT_YET_INDEXED
+            )
+
+        if status.index_status == database.IndexStatus.FAILED:
+            return SecurityInformationLookupResult.with_status(ScanLookupStatus.FAILED_TO_INDEX)
+        if status.index_status == database.IndexStatus.UNSUPPORTED:
+            return SecurityInformationLookupResult.with_status(ScanLookupStatus.UNSUPPORTED_FOR_INDEXING)
+        if status.index_status in [database.IndexStatus.TIMED_OUT, database.IndexStatus.WAITING, database.IndexStatus.IN_PROGRESS]:
+            return SecurityInformationLookupResult.with_status(ScanLookupStatus.NOT_YET_INDEXED)
+
+        assert status.index_status == database.IndexStatus.COMPLETED
+
+        # TODO(alecmerdler): Fetch data from Clair API
+        return SecurityInformationLookupResult.with_status(ScanLookupStatus.NOT_YET_INDEXED)
+
+
+    def perform_indexing(self, start_token=None):
+        whitelisted_namespaces = self.app.config.get("SECURITY_SCANNER_V4_INDEXING_WHITELIST", [])
+
+        def id_for(id_fn):
+            return (database.Manifest
+                              .select(id_fn, database.Repository, database.User)
+                              .join(database.Repository)
+                              .join(database.User)
+                              .where(database.User.username << whitelisted_namespaces if len(whitelisted_namespaces) else True)
+                              .scalar())
+
+        min_id = id_for(fn.Min(database.Manifest.id))
+        max_id = id_for(fn.Max(database.Manifest.id))
+        
+        if max_id is None or min_id is None or min_id > max_id:
+            return None
+
+        # FIXME(alecmerdler): Filter out any `Manifests` that are still being uploaded
+        def not_indexed_query():
```

Comment:
@josephschorr What do you think of these three queries? Are they accurate?

Reply:
Import the tables from database: `from data.database import Manifest` and these need the namespace filter added.

Reply:
Done